source/vibe/web/rest.d

```diff
@@ -79,16 +80,29 @@ URLRouter registerRestInterface(TImpl)(URLRouter router, TImpl instance, RestInt
 		router.registerRestInterface!T(__traits(getMember, instance, fname)(), intf.subInterfaces[i].settings);
 	}
 
+
 	foreach (i, func; intf.RouteFunctions) {
 		auto route = intf.routes[i];
 
 		// normal handler
-		auto handler = jsonMethodHandler!(func, i)(instance, intf);
+		auto handler = jsonMethodHandler!(func, i)(instance, intf, settings);
```

Comment:
There is `intf.settings`, so `settings` doesn't need to be passed here as an additional parameter. This also has the advantage of always being `!is null`, so that all `null` checks can be removed.